src/spetlr/etl/loaders/scd2_loader.py

```diff
@@ -0,0 +1,116 @@
+                   These rows (with the given join_cols)
+                   Should have "rerunned" their SCD2 values
+        time_col: The column used for creating the SCD2 logic.
+        hash_value_col: The name of the column where the sha256 encoding of the data
+
+
+        """
+        super().__init__(dataset_input_keys=dataset_input_keys)
+        self.sink_handle = sink_handle
+        self.join_cols = join_cols
+        self.time_col = time_col
+        self.hash_value_col = hash_value_col
+
+    def save(self, df: DataFrame) -> None:
+
+        _scd2_cols = ["ValidFrom", "ValidTo", "IsCurrent"]
+        _cols_without_scd2 = [col for col in df.columns if col not in _scd2_cols]
+
+        # assert _cols_without_scd2 = sink schema cols
+
+        # Generate sha56 hash value
+        # This is used for the merge statement
+        # The hash-value will be the primary key for merging
+        df = df.withColumn(
+            self.hash_value_col, f.sha2(f.concat_ws("||", *_cols_without_scd2), 256)
+        )
+
+        # Adding the hash-value column to the
+        # list over non-scd2 columns
+        # This will be used when unioning the
+        # input dataframe and the relevant data from the sink table
+        _cols_without_scd2 = _cols_without_scd2 + [self.hash_value_col]
+
+        # Selecting data from sink table
+        # it is neccesary to get the sink data.
+        # Find the rows that have the same join column values
+        # as the ones from the source table
+        # These values could potentially be updated
+        # Some Iscurrent=true rows could potential be isvcurrent=false
+        # also, if any data is late arrivals
+        # the validto and validfrom could also gets updated
+        _sink_table_name = self.sink_handle.get_tablename()
+
+        df_sink_data = self.sink_handle.read()
+
+        # This ensures, that the sink table is filtered
+        # such that is only contains rows
+        # that matches with the values in the df
+        # values found in the join columns
+        for col in self.join_cols:
+            _filter_id = df.select(col).distinct()
+            df_sink_data = df_sink_data.join(_filter_id, on=col)
+
+        # The affected sink data will now be added to the
+        # incoming dataframe.
+        df_ready = df.select(_cols_without_scd2).unionByName(
+            df_sink_data.select(_cols_without_scd2)
+        )
+
+        # Generate the SCD2 data
+        df = ValidFromToTransformer(
+            time_col=self.time_col, wnd_cols=self.join_cols
+        ).process(df_ready)
+
+        print(
+            "WARNING: This code uses a static table. "
+            "Please consider using a temporary table."
+        )
+
+        # Save the data as a view such that the merge statement
+        # can pick it up
+        temp_view_name = get_unique_tempview_name()
+        # df.createOrReplaceGlobalTempView(temp_view_name) # The correct solution
+        df.write.saveAsTable(temp_view_name)
```

Comment:
please help me understand why a temp view breaks the logic? The merge statement cannot "pick the temp view up"

Reply:
Does the function `get_unique_tempview_name()` return a string in the form of `database.some_name` or just `some_name`. What error are you getting when calling createOrReplaceGlobalTempView. And maybe it would be fine to use the non-global version as i assume this spark context will also handle the subsequent merge operation.

Reply:
it returns a uuid string: "source_98w39898574987". Just a string. Not data.some_name.
it states "source_983789479487" (or similar) does not exists!